community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/IndexingService.java

```diff
@@ -308,9 +314,65 @@ public void accept( Long indexId, IndexProxy proxy )
         samplingController.recoverIndexSamples();
         samplingController.start();
 
+        // So at this point we've started population of indexes that needs to be rebuilt in the background.
+        // Indexes backing uniqueness constraints are normally built within the transaction creating the constraint
+        // and so we shouldn't leave such indexes in a populating state after recovery.
+        // This is why we now go and wait for those indexes to be fully populated.
```

Comment:
Currently `RemoveOrphanConstraintIndexes` happens after the `start()` phase of both `IndexingService` and `Recovery`, if we could move that action to happen after `init()` of these two services, but before `start()` of them, that would mean that such indexes would be gone, and any index we wait for would be one that is supposed to be waited for.